android/src/main/java/com/swmansion/rnscreens/SearchBarView.kt

```diff
@@ -110,10 +114,12 @@ class SearchBarView(reactContext: ReactContext?) : ReactViewGroup(reactContext)
 
     private fun handleClose() {
         sendEvent(SearchBarCloseEvent(surfaceId, id))
+        setToolbarElementsVisibility(VISIBLE)
```

Comment:
Is there a possibility that this won't be fired, for example is we navigated when the search bar was open?

Reply:
@WoLewicki I thought about that already and I'm unable to reproduce the scenario where the user comes back from the search bar and in the same action comes to the previous screen. Is it even possible? 🤔 

https://github.com/software-mansion/react-native-screens/assets/23281839/30b1bade-8274-41fb-bc6e-9e972c3ca8e2

Reply:
So, I figured out how to do that (I'm calling `navigation.navigate('Main')` on `onClose`) and it's doing fine!

https://github.com/software-mansion/react-native-screens/assets/23281839/d08323fe-8b26-4ae1-aa2a-d3b839432097

Reply:
I think a better example would be to navigate forward and see what happens when you go back.

Reply:
@WoLewicki So I've checked the scenario that you've suggested and looks like everything still works - I can't manage to have search bar closed when I'm going to the next screen and coming to the previous one.

https://github.com/software-mansion/react-native-screens/assets/23281839/269faa3d-1e8d-433d-b675-83c38a9ec3fe

I'm still thinking if changing the visibility of the element could still break at some point. Do you think we should correct the change just to make sure that we will avoid side-effects?

Reply:
I think it would be some weird edge case and we can leave it as it is for now, and if the issues arise in the future, we will fix it then.